frontend/src/app/App.test.tsx

```diff
@@ -230,311 +239,382 @@ describe("App", () => {
-    wrapper.setProps(
-      getProps({
-        hostCommunication: getHostCommunicationProp({
-          currentState: getHostCommunicationState({ forcedModalClose: true }),
-          onModalReset,
-        }),
+  describe("handles HostCommunication messaging", () => {
+    let dispatchEvent: (type: string, event: Event) => void
+    let props: Props
+    let wrapper: ShallowWrapper
+    let instance: App
+    let sendMessageFunc: jest.SpyInstance
+
+    beforeEach(() => {
+      dispatchEvent = mockEventListeners()
+      props = getProps()
+      wrapper = shallow(<App {...props} />)
+      instance = wrapper.instance() as App
+
+      // @ts-expect-error - hostCommunicationMgr is private
+      instance.hostCommunicationMgr.setAllowedOriginsResp({
+        allowedOrigins: ["https://devel.streamlit.test"],
+        useExternalAuthToken: false,
       })
-    )
-    expect(wrapper.find(Modal)).toHaveLength(0)
-    expect(onModalReset).toBeCalled()
-  })
 
-  it("changes scriptRunState and fires withHostCommunication callback when scriptStopRequested signal has been received", () => {
-    const wrapper = shallow(<App {...getProps()} />)
-    const instance = wrapper.instance() as App
+      sendMessageFunc = jest.spyOn(
+        // @ts-expect-error
+        instance.hostCommunicationMgr,
+        "sendMessageToHost"
+      )
+    })
 
-    instance.isServerConnected = jest.fn().mockReturnValue(true)
+    it("sends SCRIPT_RUN_STATE_CHANGED signal to the host when the app is first rendered", () => {
+      instance.componentDidMount()
 
-    // We explicitly set the scriptRunState to RUNNING, so we can test that
-    // scriptStopRequested is handled correctly.
-    wrapper.setState({
-      scriptRunState: ScriptRunState.RUNNING,
+      expect(sendMessageFunc).toHaveBeenCalledWith({
+        type: "SCRIPT_RUN_STATE_CHANGED",
+        scriptRunState: ScriptRunState.NOT_RUNNING,
+      })
     })
 
-    wrapper.setProps(
-      getProps({
-        hostCommunication: getHostCommunicationProp({
-          currentState: getHostCommunicationState({
-            scriptStopRequested: true,
-          }),
-        }),
+    it("sends theme info to the host when the app is first rendered", () => {
+      instance.componentDidMount()
+
+      expect(sendMessageFunc).toHaveBeenCalledWith({
+        type: "SET_THEME_CONFIG",
+        themeInfo: toExportedTheme(lightTheme.emotion),
       })
-    )
+    })
 
-    expect(wrapper.state("scriptRunState")).toBe(ScriptRunState.STOP_REQUESTED)
-    expect(instance.props.hostCommunication.onScriptStop).toHaveBeenCalled()
-  })
+    it("closes modals when the modal closure message has been received", () => {
+      const dialog = StreamlitDialog({
+        type: DialogType.ABOUT,
+        sessionInfo: mockSessionInfo(),
+        onClose: () => {},
+      })
 
-  it("changes scriptRunState and fires withHostCommunication callback when scriptRerunRequested signal has been received", () => {
```

Comment:
Is this test obsolete? (I think the answer is yes, just noting where I see a test removal so I don't accidentally re-review tests that we expect to be removed.)

Reply:
Yup! Ditto to above